backend/tasks/tasks.controller.js

```diff
@@ -18,33 +18,59 @@ async function readById(req, res) {
 }
 
 async function create(req, res) {
-  const newItem = req.body;
+  /*const newItem = req.body;
 
   if (!newItem || !newItem.title) {
     return res.status(400).send("Não se esqueça do `title`!");
+  }*/
+  const { error, value } = createTaskSchema.validate(req.body);
+
+  if (error) {
+    return res.status(400).send(error.details.map(detail => detail.message).join(", "));
   }
 
-  await service.create(newItem);
-  res.status(201).send(newItem);
+  await service.create(value);
+  res.status(201).send(value);
+
+  /*await service.create(newItem);
+  res.status(201).send(newItem);*/
 }
 
 async function updateById(req, res) {
   const id = req.params.id;
-  const newItem = req.body;
+  const { error, value } = updateTaskSchema.validate(req.body);
+
+  if(error){
+    return res.status(400).send(error.details.map(detail => detail.message).join(", "));
+  }
+
+  const updatedItem = await service.updateById(id, value);
+
+  if(!updatedItem){
+    return res.status(404).send("Item não encontrado para atualização");
+  }
+  res.send(updatedItem)
+  
+  /*const newItem = req.body;
 
   if (!newItem || !newItem.title) {
     return res.status(400).send("Não se esqueça do `nome`!");
   }
 
   await service.updateById(id, newItem);
 
-  res.send(newItem);
+  res.send(newItem);*/
 }
 
 async function deleteById(req, res) {
   const id = req.params.id;
+  const result = await service.deleteById(id);
 
-  await service.deleteById(id);
+  //await service.deleteById(id);
+
+  if(!result){
```

Comment:
Espaçamento para melhorar legibilidade.
```suggestion
  if (!result) {
```